features/FEATURE_LWIP/lwip-interface/lwip-eth/arch/TARGET_Freescale/k64f_emac.c

```diff
@@ -526,15 +524,14 @@ static err_t k64f_low_level_output(struct netif *netif, struct pbuf *p)
 
   /* Wait until a descriptor is available for the transfer. */
   /* THIS WILL BLOCK UNTIL THERE ARE A DESCRIPTOR AVAILABLE */
-  while (g_handle.txBdCurrent->control & ENET_BUFFDESCRIPTOR_TX_READY_MASK)
-    osSemaphoreWait(k64f_enet->xTXDCountSem.id, osWaitForever);
+  osSemaphoreWait(k64f_enet->xTXDCountSem.id, osWaitForever);
```

Comment:
The `k64f_low_level_output` function is responsible for sending ethernet frames and is significantly detached from the socket layer. LWIP only maintains the state for non-blocking network operations, not hardware operations.

Scanning lwip's source, the return value from `k64f_low_level_output` does not appear to be respected in some cases. This makes me me think blocking is the correct behaviour, if we can recover from running out of memory.